### PR TITLE
Feature/storage path env var

### PR DESCRIPTION
**Pull request by xtuchyna**
## Related Issues and Dependencies
This PR implements one of the requirement from #132 which is a duplicate of #129 

## This introduces a breaking change

- [ ] Yes
- [X] No

## Description
Besides implementing a location environemnt variable feature, this PR also solves some of the issues detected by `pre-commit`.
However, some of the errors still remained - therefore I suggest leaving these errors for a standalone PR that would fix everything issued by `pre-commit`.

### CODE REVIEW

File: srcopsmetrics/storage.py [hunk 1]
```diff
@@ -138,19 +140,23 @@ def load_previous_knowledge(
         """Load previously collected repo knowledge. If a repo was not inspected before, create its directory.
 
         Arguments:
-            file_path {Optional[Path]} -- path to where was stored previous knowledge from
-                               inspected github repository
+            file_path {Optional[Path]} -- path to previously stored knowledge from
+                               inspected github repository. If None is passed, the used path will
+                               be the
```
Comment (KPostOffice) [inline srcopsmetrics/storage.py:145]: the used path will be the what?

File: srcopsmetrics/visualization.py [hunk 2]
```diff
@@ -472,7 +472,7 @@ def _visualize_top_x_issues_types_wrt_developers(
         )
         return fig
 
-    def _visualize_top_X_issues_types_wrt_project(
```
Comment (KPostOffice) [inline srcopsmetrics/visualization.py:475]: Is this function not used anywhere?

Comment (xtuchyna) [reply srcopsmetrics/visualization.py:475]: visualization functions will probably not be used furthermore, as we decided to move onto the Superset dashboards

therefore, as i mentioned in PR description, I suggest removing all the unused code & errors issued by `pre-commit` in other PR (in order for this PR to be more clean and specific)

any thoughts on this @pacospace ?

Comment (xtuchyna) [reply srcopsmetrics/visualization.py:475]: the 'refactor request issue' here #169

File: srcopsmetrics/storage.py (hunk 1 above)
Comment (xtuchyna) [reply srcopsmetrics/storage.py]: @KPostOffice hope it's correctly linked now?

File: srcopsmetrics/storage.py [hunk 3]
```diff
@@ -97,8 +93,13 @@ class KnowledgeStorage:
     def __init__(self, is_local: Optional[bool] = False):
         """Initialize to behave as either local or remote storage."""
         self.is_local = is_local
+        location = os.getenv(StoragePath.LOCATION_VAR.value)
```
Comment (KPostOffice) [inline srcopsmetrics/storage.py:96]: Could we add `StoragePath.DEFAULT.value` here as a default value or do we want it to fail in this case?

File: srcopsmetrics/visualization.py (hunk 2 above)
Comment (pacospace) [reply srcopsmetrics/visualization.py:475]: I think make sense to remove what we do not need anymore, although in this case someone could use the visualizations locally maybe using the one available

Comment (xtuchyna) [reply srcopsmetrics/visualization.py:475]: yes, but you can also use Superset locally, right?

File: srcopsmetrics/storage.py (hunk 3 above)
Comment (xtuchyna) [reply srcopsmetrics/storage.py:96]: yes, this is true, no need to fail if we still have a default path

Comment (KPostOffice) [inline srcopsmetrics/storage.py:96]: `os.getenv` allows you to indicate a default value like this: `os.getenv(<envvar>, default)`

Comment (xtuchyna) [reply srcopsmetrics/storage.py]: nice one! thx!

**Comment by KPostOffice** [comment]
@pacospace

File: srcopsmetrics/visualization.py (hunk 2 above)
Comment (pacospace) [reply srcopsmetrics/visualization.py:475]: Superset locally?

**Pull request merged**